File: leetcode/maximum-subarray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSubArray = function(nums) {
    
    let ans = nums[0];
    let subSum = 0;
        
    for(let i=0;i<nums.length;i++) {  
        subSum += nums[i];  
        ans = Math.max(subSum, ans);
        
        if(subSum < 0)
            subSum = 0;
    }
        
    return ans;
};

File: leetcode/maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        
        ans = nums[0]
        sub_sum = 0
        
        for num in nums:
            sub_sum += num
            ans = max(sub_sum, ans)
            
            if sub_sum < 0:
                sub_sum = 0
        
        return ans
        
